firestore.rules

```diff
@@ -1,77 +1,56 @@
 rules_version = '2';
 service cloud.firestore {
   match /databases/{database}/documents {
-    // Bucket Rules
-    match /buckets/{bucketId} {
-      // Any autenticated can create a bucket
-      allow create: if request.auth != null;
-      // But only the owner can read, update, and delete it
-      allow read, write, update, delete: if request.auth != null && request.auth.uid == resource.data.owner;
+    // High level rules
+    function userIsOwner(owner) {
+      // Ensure the User ID matches the Auth UID
+      return owner == request.auth.uid;
     }
-    // Key Rules
-    match /accessKeys/{keyId} {
-      // Any autenticated can create a key
-      allow create: if request.auth != null;
-      // Allow to read with the keyId as the uid (for reading from service account)
-      allow read: if request.auth.uid == keyId || request.auth.uid == resource.data.owner 
-      // But only the owner can read, update, and delete it
-      allow write, update, delete: if request.auth != null && request.auth.uid == resource.data.owner;
+
+    function fingerprintIsValid(fingerprint) {
+      // Make sure the ID is a 16 character hex string
+      return fingerprint.matches("^[0-9a-f]{16}$");
```

Comment:
We definitely want to store the entire fingerprint, trimming hashes down in git is fine, or pointing at unique references but not in crypto land. The hashes aren't going to be user-visible so this also isn't an ease-of-use argument.

Reply:
heard

Reply:
fingerprint utility no longer truncates the digest
firestore rule checks for length of 40 hex chars, since the sha1 digests are 20 bytes